backend/api/deck.go

```diff
@@ -41,38 +44,206 @@ func (a *API) getDeckHandler(c *gin.Context) {
 		c.JSON(404, gin.H{"error": "deck not found"})
 		return
 	}
-	d, err := decompressDeck(deck)
+
+	resBts, err := deck.Bytes()
 	if err != nil {
 		c.JSON(500, gin.H{"error": err.Error()})
 		return
 	}
 
-	keys := make([]string, 0, len(d))
-	for k := range d {
-		keys = append(keys, k)
+	c.Data(200, "text/plain", resBts)
+}
+
+// region deck parsing unstable
+
+// deck designed to be parsed once and then used for lookups. The load of the parsing is in the request context as to
+// not block the sub
+type deck struct {
+	// raw data not parsed, after parsing, raw data buf is freed and replaced by ready-to-use result
+	buf []byte
+
+	// ensures work is only done once even when racing for deck parse. Once its loaded, will be checked
+	// using atomic.LoadUint32 instead of a mutex lock
+	parseOnce sync.Once
+}
+
+// Bytes use to get parsed deck
+func (d *deck) Bytes() (res []byte, err error) {
```

Comment:
Right now it is fairly obvious when the expensive part of this happens: is it worth doing something (maybe even just renaming this) to make it clear this is the expensive call?